redis-utils/src/main/java/org/ballerinalang/redis/actions/AbstractRedisAction.java

```diff
@@ -1576,29 +1514,33 @@ private BValueArray createBStringArrayFromList(List<String> list) {
         return bMap;
     }
 
-    protected String[] createArrayFromBStringArray(BValueArray bStringArray) {
-        String[] stringArray = new String[(int) bStringArray.size()];
-        for (int i = 0; i < bStringArray.size(); i++) {
-            stringArray[i] = bStringArray.getString(i);
+    static String[] createStringArrayFromBArray(BArray bStringArray) {
+        String[] stringArray = new String[bStringArray.size()];
+        int i = 0;
+        for (Object item : bStringArray.getStringArray()) {
+            stringArray[i++] = item.toString();
         }
         return stringArray;
     }
 
-    protected Map<String, String> createMapFromBMap(BMap<String, BString> bMap) {
-        Map<String, String> map = new HashMap<>();
-        bMap.getMap().keySet().forEach(item -> map.put(item, bMap.get(item).stringValue()));
+    static Map<String, Object> createMapFromBMap(BMap<String, Object> bMap) {
+        Map<String, Object> map = new LinkedHashMap<>();
+        for (Map.Entry<String, Object> entry : bMap.entrySet()) {
+            map.put(entry.getKey(), entry.getValue());
+        }
         return map;
     }
 
-    protected void setNullableReturnValues(BValue result, Context context) {
-        if (result == null) {
-            context.setReturnValues();
-        } else {
-            context.setReturnValues(result);
+    static BArray createBstringArrayFromBMap(BMap<String, BString> bMap) {
+        BArray bStringArray = new ArrayValueImpl(new String[bMap.size()]);
+        int i = 0;
+        for (Map.Entry<String, BString> entry : bMap.entrySet()) {
+            bStringArray.add(i++, entry.getValue().stringValue());
```

Comment:
We can use `bStringArray.append()` in here too.

Reply:
Updated in a6ce8d8